pkgs/development/interpreters/gnudatalanguage/default.nix

```diff
@@ -0,0 +1,156 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, wrapGAppsHook
+, readline
+, ncurses
+, zlib
+, gsl
+, openmp
+, graphicsmagick
+, netcdf
+, hdf4
+, hdf5
+, fftw
+, fftwFloat
+, fftwLongDouble
+, proj
+, shapelib
+, expat
+, udunits
+, eigen
+, pslib
+, eccodes
+, glpk
+, libpng
+, plplot
+, libtiff
+, libgeotiff
+, libjpeg
+, enableLibtirpc ? stdenv.isLinux, libtirpc
+, pythonInterpreter ? python3, python3
+, enableMPI ? (stdenv.isLinux || stdenv.isDarwin)
+# Choose mpich because of it builds on aarch and darwin
+, libraryMPI ? mpich
+, mpich
+# Unfree optional dependency
+# for hdf4 and hdf5
+, enableSzip ? false, szip
+# wxWidgets is preferred over X11 for this project
+# but we only have it on Linux and Darwin in Nixpkgs
+, enableWX ? (stdenv.isLinux || stdenv.isDarwin)
+, wxGTK31, wxmac
+# X11: OFF by default for platform consistency
+# Use X where WX is not available
+, enableXWin ? (!stdenv.isLinux && !stdenv.isDarwin)
+, xorg
+}:
+
+let
+  wxWidgets = (if stdenv.isDarwin then wxmac else wxGTK31);
+  enablePlplotDrivers = enableWX || enableXWin;
+  hdf4-custom = (hdf4.override (
+    {
+      uselibtirpc = enableLibtirpc;
+    } // (lib.optionalAttrs enableSzip {
+      inherit szip;
+    })
+  ));
+  hdf5-custom = (hdf5.override (
+    (lib.optionalAttrs enableMPI {
+      mpiSupport = true;
+      mpi = libraryMPI;
+    }) // (lib.optionalAttrs enableSzip {
+      inherit szip;
+    })
+  ));
+  plplot-with-drivers = (plplot.overrideAttrs (oldAttrs: {
+    buildInputs = (oldAttrs.buildInputs or []
+      ++ lib.optional enableWX wxWidgets
+      ++ lib.optional enableXWin xorg.libX11
+    );
```

Comment:
Same here, it'd be great to edit the `plplot` expression so you could override it here simply with `inherit enableWx enableXWin` and you'd add `wxWidgets` and `xorg.libX11` to a passthru there, so you'd use the same dependencies here.